arcgis-ios-sdk-samples/Edit data/Delete features (feature service)/DeleteFeaturesViewController.swift

```diff
@@ -0,0 +1,147 @@
+// Copyright 2015 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class DeleteFeaturesViewController: UIViewController, AGSMapViewTouchDelegate, AGSCalloutDelegate {
+    
+    @IBOutlet private var mapView:AGSMapView!
+    
+    private var featureTable:AGSServiceFeatureTable!
+    private var featureLayer:AGSFeatureLayer!
+    private var lastQuery:AGSCancellable!
+    private var selectedFeature:AGSFeature!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        //add the source code button item to the right of navigation bar
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["DeleteFeaturesViewController"]
+        
+        //instantiate map with a basemap
+        let map = AGSMap(basemap: AGSBasemap.streetsBasemap())
+        //set initial viewpoint
+        map.initialViewpoint = AGSViewpoint(center: AGSPoint(x: 544871.19, y: 6806138.66, spatialReference: AGSSpatialReference.webMercator()), scale: 2e6)
+        
+        //assign the map to the map view
+        self.mapView.map = map
+        //set touch delegate on map view as self
+        self.mapView.touchDelegate = self
+        
+        //instantiate service feature table using the url to the service
+        self.featureTable = AGSServiceFeatureTable(URL: NSURL(string: "http://sampleserver6.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer/0"))
+        //create a feature layer using the service feature table
+        self.featureLayer = AGSFeatureLayer(featureTable: self.featureTable)
+        
+        //add the feature layer to the operational layers on map
+        map.operationalLayers.addObject(featureLayer)
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    func showCallout(feature:AGSFeature) {
+        let title = feature.attributeValueForKey("typdamage") as! String
+        self.mapView.callout.title = title
+        self.mapView.callout.delegate = self
+        self.mapView.callout.accessoryButtonImage = UIImage(named: "Discard")
+        self.mapView.callout.showCalloutForFeature(feature, layer: self.featureLayer, tapLocation: AGSPoint(x: 0, y: 0, spatialReference: self.mapView.map!.spatialReference), animated: true)
```

Comment:
Why is the tapLocation 0,0? Shouldn't it be nil, or the actual location the user tapped?